pkg/kvstore/redis.go

```diff
@@ -162,13 +169,50 @@ func (s *redisKvStore) PutBatch(ctx context.Context, values interface{}) error {
 		return fmt.Errorf("could not convert values from %T to map[interface{}]interface{}", values)
 	}
 
+	chunkSize := s.settings.BatchSize
+	pairs := make([]interface{}, 0, chunkSize)
```

Comment:
It is nice that you try to already reserve enough capacity, but then you should reserve twice that much because you add keys and values.

Reply:
done